mmaction/apis/test.py

```diff
@@ -107,7 +107,7 @@ def collect_results_cpu(result_part, size, tmpdir=None):
                                 dtype=torch.uint8,
                                 device='cuda')
         if rank == 0:
-            tmpdir = tempfile.mkdtemp()
+            tmpdir = tempfile.mkdtemp(dir='.dist_test')
```

Comment:
That would be nice. BTW, I would like to mention that there is still risk in this version if you test multiple models at the same time, i.e. the middle results of different test runs will mess up. What about generating a hash based on the testing config and testing checkpoint, then use it as the name of the tmpdir.

Reply:
The case you mentioned is for `tmpdir is not None`, which is for Line#116

Reply:
I think, actually, the two cases are the same. If `tmpdir` is set, testing can run well when running on multiple nodes. But no user would like to designate `tmpdir` every time. If we can generate a tmpdir with hashing as its name each time, some efforts would be saved. Anyway, this PR is a good step.

Reply:
How about use time-hash for tmpdir name, since it is difficult to obtain testing config and testing checkpoint here.

Reply:
> if you test multiple models at the same time

You mean testing multiple models in a same session?

Reply:
my understanding is that `mkdtemp` will create a unique folder name?

Reply:
> How about use time-hash for tmpdir name, since it is difficult to obtain testing config and testing checkpoint here.

I'm not sure, since we have multiple processes running at the same time, maybe they will get different hashes.

Reply:
> my understanding is that `mkdtemp` will create a unique folder name?

That's right. The problem is, by default it creates a tempdir under /tmp, while for different nodes on HPC, they all have their own /tmp.

Reply:
> > if you test multiple models at the same time
> 
> You mean testing multiple models in a same session?

At the same time & on the same storage system.

Reply:
`tempfile.mkdtemp(dir='.dist_test')` will create a temp dir with a random name under `.dist_test`, not under `/tmp`.